src/shared_modules/content_manager/tests/unit/fileDownloader_test.cpp

```diff
@@ -0,0 +1,241 @@
+/*
+ * Wazuh content manager - Unit Tests
+ * Copyright (C) 2015, Wazuh Inc.
+ * Jun 08, 2023.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#include "fileDownloader_test.hpp"
+#include "fakes/fakeServer.hpp"
+#include "fileDownloader.hpp"
+#include "json.hpp"
+#include "updaterContext.hpp"
+#include "gtest/gtest.h"
+#include <filesystem>
+#include <memory>
+#include <stdexcept>
+
+const auto OK_STATUS = R"({"stage":"FileDownloader","status":"ok"})"_json;
+const auto FAIL_STATUS = R"({"stage":"FileDownloader","status":"fail"})"_json;
+const auto BASE_URL = "localhost:4444/";
+const auto CONTENT_FILE_NAME = "content.xyz";
+
+void FileDownloaderTest::SetUpTestSuite()
+{
+    if (!m_spFakeServer)
+    {
+        m_spFakeServer = std::make_unique<FakeServer>("localhost", 4444);
+    }
+}
+
+void FileDownloaderTest::TearDownTestSuite()
+{
+    m_spFakeServer.reset();
+}
+
+void FileDownloaderTest::SetUp()
+{
+    m_spUpdaterBaseContext = std::make_shared<UpdaterBaseContext>();
+    m_spUpdaterBaseContext->downloadsFolder = (m_outputFolder / "downloads").string();
+    m_spUpdaterBaseContext->contentsFolder = (m_outputFolder / "contents").string();
+    m_spUpdaterBaseContext->configData["url"] = BASE_URL;
+    m_spUpdaterBaseContext->configData["contentFileName"] = CONTENT_FILE_NAME;
+
+    m_spUpdaterContext = std::make_shared<UpdaterContext>();
+    m_spUpdaterContext->spUpdaterBaseContext = m_spUpdaterBaseContext;
+
+    std::filesystem::create_directory(m_outputFolder);
+    std::filesystem::create_directory(m_spUpdaterBaseContext->downloadsFolder);
+    std::filesystem::create_directory(m_spUpdaterBaseContext->contentsFolder);
+}
+
+void FileDownloaderTest::TearDown()
+{
+    std::filesystem::remove_all(m_outputFolder);
+}
+
+/**
+ * @brief Tests the instantiation of the class.
+ *
+ */
+TEST_F(FileDownloaderTest, Instantiation)
+{
+    EXPECT_NO_THROW(std::make_shared<FileDownloader>());
+    EXPECT_NO_THROW(FileDownloader());
+}
+
+/**
+ * @brief Tests the downloader with an invalid URL.
+ *
+ */
+TEST_F(FileDownloaderTest, DownloadBadURL)
+{
+    // Set up expected data.
+    nlohmann::json expectedData;
+    expectedData["paths"] = m_spUpdaterContext->data.at("paths");
+    expectedData["stageStatus"] = nlohmann::json::array();
+    expectedData["stageStatus"].push_back(FAIL_STATUS);
+
+    // Set invalid config data. This will make the downloader to download from 'localhost:4444/invalid_file'.
+    m_spUpdaterBaseContext->configData["compressionType"] = "raw";
+    m_spUpdaterBaseContext->configData["url"].get_ref<std::string&>() += "invalid_file";
```

Comment:
Total nitpick:
How about..
```suggestion
    m_spUpdaterBaseContext->configData["url"] = BASE_URL + "invalid_file";
```
(BASE_URL must be made a std::string)

Reply:
Done! Much cleaner.